File: Backend/DigitalStore.Domain/Entities/Topic.cs
using DigitalStore.Domain.Common;
using System.Collections.Generic;

namespace DigitalStore.Domain.Entities
{
    public class Topic
    {
        public int Id { get; set; }
        public int? ParentId { get; set; }
        public required string Title { get; set; }
        public string? ImageUrl { get; set; }

        public Topic? Parent { get; set; }
        public ICollection<Topic> Children { get; set; } = new List<Topic>();
        public ICollection<PackageTopic> PackageTopics { get; set; } = new List<PackageTopic>();
    }
}

File: education_app/lib/features/topics/domain/entities/topic.dart
import 'package:equatable/equatable.dart';

class Topic extends Equatable {
  final int id;
  final int? parentId;
  final String title;
  final String? imageUrl;
  final List<Topic> children;

  const Topic({
    required this.id,
    this.parentId,
    required this.title,
    this.imageUrl,
    this.children = const [],
  });

  @override
  List<Object?> get props => [id, parentId, title, imageUrl, children];
}
